src/java.base/share/classes/java/lang/Class.java

```diff
@@ -314,6 +314,14 @@ public String toGenericString() {
                     sb.append(' ');
                 }
 
+                // A class cannot be strictfp and sealed so it is
+                // sufficient to check for sealed-ness after all
+                // modifiers are printed.
+                boolean isSealed = isSealed();
```

Comment:
A small concern about performance.  
Class.isSealed() calls getPermittedSubclasses() which assembles and filters the list of classes and a possible security manager check and then only checks for non-null.
Perhaps a separate issue could look at that.

Reply:
If Class.toGenericString() turned out to be performance sensitive, which would be a surprise, then we could look into alternate implementations.